packages/core/src/Utils/Analytics/analytics.ts

```diff
@@ -1,48 +1,184 @@
+        const fnv1aHash = (string: string): number => {
+            let hash = 0x811c9dc5;
+            for (let i = 0; i < string.length; i++) {
+                // eslint-disable-next-line no-bitwise
+                hash ^= string.charCodeAt(i);
+                // eslint-disable-next-line no-bitwise
+                hash = (hash * 0x01000193) >>> 0;
+            }
+            return hash;
+        };
+        const base64Encode = (string: string): string => btoa(string);
+        const hash = fnv1aHash(inputString).toString(16);
+        let combined = base64Encode(hash);
+        while (combined.length < desiredLength) {
+            combined += base64Encode(fnv1aHash(combined).toString(16));
+        }
+        return combined.substring(0, desiredLength);
+    },
+    getCookies: (name: string): string | null => {
+        const value = `; ${document.cookie}`;
+        const parts = value.split(`; ${name}=`);
+        if (parts.length === 2) {
+            const part = parts.pop();
+            const cookieValue = part ? decodeURIComponent(part.split(';').shift()!) : null;
+            try {
+                return cookieValue ? JSON.parse(cookieValue) : null;
+            } catch (e) {
+                return cookieValue;
+            }
+        }
+        return null;
+    },
+    trackPageUnload: () => {
+        window.addEventListener('beforeunload', event => {
+            if (!cacheTrackEvents.isPageViewSent()) {
+                cacheTrackEvents.push('cached_analytics_page_views', {
+                    name: window.location.href,
+                    properties: {
+                        url: window.location.href,
+                    },
                 });
-
-                eventQueue = [];
-                localStorage.removeItem('pending_events');
             }
+        });
+    },
+    isReady: (): boolean => {
+        if (typeof Analytics === 'undefined' || Analytics === null) {
+            return false;
         }
-    } catch (error) {
-        // eslint-disable-next-line no-console
-        console.log(error);
-    }
-};
-
-const setEvent = (event: AnalyticsEvent): void => {
-    const storedEvents = localStorage.getItem('pending_events');
-    let eventQueue: AnalyticsEvent[] = [];
-    if (storedEvents) {
-        eventQueue = JSON.parse(storedEvents) as AnalyticsEvent[];
-    }
-    eventQueue.push(event);
-    localStorage.setItem('pending_events', JSON.stringify(eventQueue));
-};
-
-const trackEventWithCache = (event: AnalyticsEvent): void => {
-    if (window.rudderanalytics) {
-        handleCachedEvents();
-        window.rudderanalytics.track(event.name, event.properties);
-    } else {
-        setEvent(event);
-    }
+        const instances = Analytics?.getInstances();
+        return !!instances?.tracking;
+    },
+    parseCookies: (cookieName: string): Event[] | null => {
+        const cookies = document.cookie.split('; ').reduce((acc: Record<string, string>, cookie) => {
```

Comment:
Do we need parseCookies?We can get the cookies value from Cookies

Reply:
Fixed.